util/src/main/java/io/kubernetes/client/util/leaderelection/resourcelock/EndpointsLock.java

```diff
@@ -0,0 +1,115 @@
+package io.kubernetes.client.util.leaderelection.resourcelock;
+
+import io.kubernetes.client.apis.CoreV1Api;
+import io.kubernetes.client.models.V1Endpoints;
+import io.kubernetes.client.models.V1ObjectMeta;
+import io.kubernetes.client.util.leaderelection.LeaderElectionRecord;
+import io.kubernetes.client.util.leaderelection.Lock;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicReference;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class EndpointsLock implements Lock {
```

Comment:
I feel like this duplicates a lot of code with the `ConfigMapLock` does it make sense to refactor a parent class for both of these? `AbstractResourceLock` or somesuch?

Reply:
it's helping reducing code redundancy tho but having an `AbstractResourceLock` is a bit overkilling i suppose. for now, configmap and endpoints are the only supported resources and we aren't leaning to extend more. also, soon future on, the `coordination.Lease` API will be replacing the two.